server/service/appconfig.go

```diff
@@ -305,6 +322,23 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte) (*fleet.AppCo
 	}
 	appConfig.Integrations.Zendesk = newZendeskConfig
 
+	transparencyURL := appConfig.FleetDesktop.TransparencyURL
+	if transparencyURL == "" {
+		transparencyURL = defaultTransparencyURL
+	}
+
+	if transparencyURL != defaultTransparencyURL && license.Tier != "premium" {
+		invalid.Append("transparency_url", "requires Fleet Premium license")
+		return nil, ctxerr.Wrap(ctx, invalid)
+	}
+
+	if _, err := url.Parse(transparencyURL); err != nil {
```

Comment:
Do we want any validation of the url itself? `url.Parse` doesn't really provide much validation so I'm not sure we're getting any real value out of this check.